File: cross-platform/dual-reader/dual_reader/lib/src/domain/repositories/bookmark_repository.dart
import 'package:dual_reader/src/domain/entities/bookmark_entity.dart';

abstract class BookmarkRepository {
  Future<List<BookmarkEntity>> getBookmarks(String bookId);
  Future<BookmarkEntity> addBookmark(BookmarkEntity bookmark);
  Future<void> deleteBookmark(String id);
  Future<BookmarkEntity> updateBookmark(BookmarkEntity bookmark);
}

File: cross-platform/dual-reader/dual-reader-android/app/src/main/java/com/dualreader/app/domain/repositories/BookmarkRepository.kt
package com.dualreader.app.domain.repositories

import com.dualreader.app.domain.entities.Bookmark
import kotlinx.coroutines.flow.Flow

interface BookmarkRepository {
    fun getBookmarksForBook(bookId: String): Flow<List<Bookmark>>
    suspend fun addBookmark(bookmark: Bookmark)
    suspend fun deleteBookmark(id: String)
    suspend fun deleteBookmarksForBook(bookId: String)
}
